tests/Test.ADAL.NET.Integration/AcquireTokenSilentTests.cs

```diff
@@ -106,5 +108,59 @@ public void AcquireTokenSilentServiceErrorTestAsync()
             Assert.IsNotNull(result);
             Assert.AreEqual("existing-access-token", result.AccessToken);
         }
+
+        [TestMethod]
+        [TestCategory("AcquireTokenSilentTests")]
+        //292916 Ensure AcquireTokenSilent tests exist in ADAL.NET for public clients
+        public void AcquireTokenSilentWithEmptyCacheTest()
+        {
+            var context = new AuthenticationContext(TestConstants.DefaultAuthorityHomeTenant, true, new TokenCache());
+            AuthenticationResult result;
+            AdalSilentTokenAcquisitionException ex = AssertException.TaskThrows<AdalSilentTokenAcquisitionException>(async () => 
+            result = await context.AcquireTokenSilentAsync(TestConstants.DefaultResource, TestConstants.DefaultClientId, new UserIdentifier("unique_id", UserIdentifierType.UniqueId)));
+
+            Assert.AreEqual(ex.ErrorCode, AdalError.FailedToAcquireTokenSilently);
+            Assert.AreEqual(ex.Message, AdalErrorMessage.FailedToAcquireTokenSilently);
+        }
+
+        [TestMethod]
+        [TestCategory("AcquireTokenSilentTests")]
+        //292916 Ensure AcquireTokenSilent tests exist in ADAL.NET for public clients
+        public void AcquireTokenSilentTestWithInvalidTokenInCache()
+        {
+            var context = new AuthenticationContext(TestConstants.DefaultAuthorityHomeTenant, new TokenCache());
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Post,
+                ResponseMessage = MockHelpers.CreateInvalidGrantTokenResponseMessage(),
+                PostData = new Dictionary<string, string>()
+                {
+                    {"client_id", TestConstants.DefaultClientId},
```

Comment:
add RefreshToken = "some-rt" as a part of the postdata validation

Reply:
@kpanwar is this what you mean?
                    { "client_id", TestConstants.DefaultClientId},
                    {"grant_type", "refresh_token"},
                    {"refresh_token", "some-rt" }

there is no property named RefreshToken to set